ambry-store/src/main/java/com/github/ambry/store/DiskManager.java

```diff
@@ -570,4 +590,138 @@ private void reportUnrecognizedDirs() {
       }
     }
   }
+
+  public DiskHealthStatus getDiskHealthStatus(){
+    return diskHealthStatus;
+  }
+
+  /**
+    Manages the disk healthchecking tasks such as creating,writing, reading, and deleting a file
+   */
+  private class DiskHealthCheck{
+    private final int operationTimeoutSeconds;      //how long a read/write operation is allotted
+    private final boolean enabled;                  //enabled healthchecking
+    private final String heathcheckDir;             //filepath of where the file will be created
+
+
+    DiskHealthCheck(int operationTimeoutSeconds, boolean enabled, String heathcheckDir){
+      this.enabled = enabled;
+      this.heathcheckDir = heathcheckDir;
+      this.operationTimeoutSeconds = operationTimeoutSeconds;
+
+    }
+
+    /**
+     * Getter utility for enabled boolean
+     * @return flag to indicate disk healthchecking or not
+     */
+    public boolean isEnabled(){
+      return enabled;
+    }
+
+    /**
+     * Attempts to close the filechannel and delete the file
+     * @param fileChannel the file used for disk healthcheck
+     * @return True/False depending on the close/delete was successful
+     */
+    private boolean deleteFile(AsynchronousFileChannel fileChannel){
+      if(fileChannel.isOpen()){
+        try{
+          fileChannel.close();
+        }catch(IOException e){
+          diskHealthStatus = diskHealthStatus.DELETE_IO_EXCEPTION;
```

Comment:
should be 
```
if (diskHealthStatus == HEALTHY) {
    diskHealthStatus = diskHealthStatus.DELETE_IO_EXCEPTION;
}
```

Reply:
I changed the if-statement to account for any issue that's happened thus far during the healthcheck so as to not overlap with that diskHealthStatus
`if (successfulThusFar) {
    diskHealthStatus = diskHealthStatus.DELETE_IO_EXCEPTION;
}`